lib/src/converter.dart

```diff
@@ -97,8 +103,16 @@ class CombineConverter<R, T> extends Converter<T> {
   ///
   /// The choices for this converter will be inherited from [converter], but can be overridden by
   /// passing [choices].
-  CombineConverter(this.converter, this.process, {Iterable<ArgChoiceBuilder>? choices})
-      : _choices = choices,
+  ///
+  /// The type for this converter will be inherited from [converter], but can be overridden by
+  /// passing [type].
+  CombineConverter(
+    this.converter,
+    this.process, {
+    Iterable<ArgChoiceBuilder>? choices,
+    CommandOptionType? type,
+  })  : _choices = choices,
+        _type = type,
```

Comment:
Thats kinda weirdm I prefer setting field late and setting stuff in constuctor body

Reply:
I would agree (and others too, see https://github.com/dart-lang/language/issues/2005) but I feel using late fields is more error prone and does not adhere to the effective dart style guide

Reply:
in that case final field or late final field its going to be the same so I dont see the issue. and code would be much more readable

Reply:
This constructor doesn't have a body anyways though, so you'd just be moving the two lines `_choices = choices` and `_type = type` into a constructor body, this doesn't really affect readability.

Reply:
in that case sure